cpp/src/arrow/util/uri.cc

```diff
@@ -58,10 +59,10 @@ bool IsDriveSpec(const std::string_view s) {
 
 }  // namespace
 
-std::string UriEscape(const std::string& s) {
+std::string UriEscape(const std::string_view& s) {
```

Comment:
Minor nit: I think we've been preferring to pass `string_view` by value in Substrait per https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/ and @bkietz 's advice

Reply:
Ah, good point, thanks.

Reply:
I opened https://github.com/apache/arrow/issues/15171 for other instances of this.